plugins/SpectrumAnalyzer/SaProcessor.cpp

```diff
@@ -650,7 +653,8 @@ float SaProcessor::ampToYPixel(float amplitude, unsigned int height) const
 	if (m_controls->m_logYModel.value())
 	{
 		// logarithmic scale: convert linear amplitude to dB (relative to 1.0)
-		float amplitude_dB = 10 * log10(amplitude);
+		assert (amplitude >= 0);
+		float amplitude_dB = 10 * log10(amplitude == 0 ? std::numeric_limits<float>::min() : amplitude);
```

Comment:
```suggestion
		assert(amplitude >= 0);
		float amplitude_dB = 10 * std::log10(std::max(amplitude, std::numeric_limits<float>::min()));
```
style + simplification

Reply:
Thanks @messmerd! That indeed looks much more concise. Done with commit 1e7aed43aa9 which also adds some optimization with regards to the call to `std::numeric_limits<float>::min()`.

Reply:
There's no need to assign the result of `std::numeric_limits<float>::min()` to a variable for optimization purposes because it's a `constexpr` function which is executed at compile time.

Reply:
Ah, you're right. I wrongfully assumed that the output at https://godbolt.org/ was optimized code. Fixed with commit 068f74374a8. The commit message contains a small snippet that can be used for experimentation with the "Compiler Explorer".